### PR TITLE
edit estate function and update query

### PR DESCRIPTION
**Pull request by muhammadharoun**

### CODE REVIEW

**Review by ElhamFadel — DISMISSED** [review]

File: server/controllers/estates/editEstate.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      category,
+      street,
+      city,
+      region,
+      bathrooms,
+      bedrooms,
+      rooms,
+      space,
+      available,
+    );
```
Comment (Mu7ammadAbed) [inline server/controllers/estates/editEstate.js:38]: You need to fix this and make it more readable

Comment (Mu7ammadAbed) [reply server/controllers/estates/editEstate.js:38]: and by the way, you can destructure them from the result directly

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]
What is the case if there was no estate with this id?
What if the database throws an error for some reason and the row didn't update and you still sending status code 200(ok)?

File: server/test/index.test.js [hunk 2]
```diff
@@ -61,3 +61,28 @@ describe('user estates', () => {
...
+//         bedrooms: 's',
+//         rooms: 's',
+//         space: 's',
+//         available: false,
+//       })
+//       .expect(200)
+//       .expect('Content-Type', /json/);
+//     console.log(res);
+//   });
+// });
```
Comment (Mu7ammadAbed) [inline server/test/index.test.js:88]: ??

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: server/controllers/estates/editEstate.js [hunk 3]
```diff
@@ -3,39 +3,21 @@ const { editEstateValidation } = require('../../utils/validation/editEstateValid
...
-      bedrooms,
-      rooms,
-      space,
-      available,
+      estateId, title, price, description,
+      type, category, street, city, region,
+      bathrooms, bedrooms, rooms, space, available,
+    } = await editEstateValidation.validateAsync({ ...req.body, ...req.params });
+
+    const editResult = await editEstateQuery(
```
Comment (Mu7ammadAbed) [inline server/controllers/estates/editEstate.js:12]:
```suggestion
    const { rowCount } = await editEstateQuery(
```

File: server/controllers/estates/editEstate.js [hunk 4]
```diff
@@ -3,39 +3,21 @@ const { editEstateValidation } = require('../../utils/validation/editEstateValid
...
+      estateId, title, price, description, type,
+      category, street, city, region, bathrooms,
+      bedrooms, rooms, space, available,
     );
+    if (editResult.rowCount === 0) {
+      res.status(406).json({ status: 406, msg: 'enter valid estate id ' });
+    }
+
     res.json({ status: 200 });
```
Comment (Mu7ammadAbed) [inline server/controllers/estates/editEstate.js]:
```suggestion
    if (rowCount === 0) {
      return res.status(400).json({ status: 400, message: 'enter valid estate id ' });
    }

    res.json({ message: 'Estate updated successfully' });
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]
You need to convert your tests to `async/await` and you need to cover more cases in your tests

File: server/controllers/users/userEstates.js [hunk 5]
```diff
@@ -1,5 +1,6 @@
 const { userEstatesQuery } = require('../../database/quieres/index');
 
+// eslint-disable-next-line consistent-return
```
[the remark this answers is not in the document]
Comment (Mu7ammadAbed) [reply server/controllers/users/userEstates.js:3]:
```suggestion
```

**Review by ElhamFadel — APPROVED** [review]

**Review by mohammedsalah7 — DISMISSED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by mohammedsalah7 — APPROVED** [review]

**Pull request merged**